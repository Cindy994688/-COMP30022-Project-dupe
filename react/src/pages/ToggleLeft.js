import React, { Component } from 'react'
import Jumbotron from 'react-bootstrap/Jumbotron'
import Profile from './Profile.js'
import UserInfo from './userInfo.js'
import axios from 'axios'
import './pages.css'
import { Button, ButtonGroup, ButtonToolbar } from 'react-bootstrap-buttons';
import 'react-bootstrap-buttons/dist/react-bootstrap-buttons.css';
import { Image } from 'react-native';
//thank you to https://www.youtube.com/watch?v=x5oiX93DeHA

export default class ToggleProj extends Component {
    state = {
        cv: false,
        personalproj: false,
    }

    constructor(props) {
      super(props);
      this.state = {
        output: "",
        fullname:"",
      };
    }

    componentDidMount() {
      const name = axios.get('/name/'+this.props.name)
        .then((res)=>{
          this.setState({
            fullname: res.data,
          })
        });
    }


    togglecv = () => {
        this.setState({
            cv: !this.state.cv,
            personalproj: false,
        })
    }

    togglepersonalproj = () => {
        this.setState({
            cv: false,
            personalproj: !this.state.personalproj
        })
    }


    render() {
        return (
            <div className = "vertBox">
                <div className = "vertBox">
                  {/*<div  style={{display: 'inline-block'}}>*/}
                  <div className = "box">
                    <div>
                      <Profile img = {'/image/'+this.props.name+'.jpg'} clientname = {this.state.fullname}/>
                    </div>
                    {/*<div style={{display: 'inline-block'}}>*/}
                    <div className = "vertBox">
                    <p><UserInfo clientname = {this.props.name} type = "description"/></p>
                    <p><UserInfo clientname = {this.props.name} type = "skills"/></p>
                    <p><UserInfo clientname = {this.props.name} type = "email"/></p>
                    </div>
                  </div>
                  <div className="buttondiv">
                  <button type="submit" className="btn btn-primary button" onClick = {this.togglecv}  style={{cursor:'pointer'}}>See CV</button>
                  <button type="submit" className="btn btn-primary button" onClick = {this.togglepersonalproj} style={{cursor:'pointer'}}>See Personal Projects</button>
                  </div>
                </div>

                <div>
                {this.state.cv &&
                  <div className = "box">
                  <Image source={"/image/MengyanResume.jpg"}
                      style={{width:636, height:900}}/>
                  </div>
                }

                </div>

                <div>
                {this.state.personalproj &&
                    <div className = "box">
                    <p>This is a 100% verified real personal project</p>
                    </div>}
                </div>

            </div>
        )
    }
}
