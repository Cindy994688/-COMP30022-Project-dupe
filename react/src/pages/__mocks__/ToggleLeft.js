import React, { Component } from 'react'
import Profile from './Profile.js'
import UserInfo from './userInfo.js'
import axios from 'axios'
import './pages.css'
import './darkmode.css'
import './access.css'
import 'react-bootstrap-buttons/dist/react-bootstrap-buttons.css';
//thank you to https://www.youtube.com/watch?v=x5oiX93DeHA
import ProjectsPage from './ProjectsPage';

export default class ToggleProj extends Component {
    state = {
        cv: false,
        personalproj: false,
    }

    constructor(props) {
      super(props);
      this.state = {
        output: "",
        fullname:"",
      };
    }

    componentDidMount() {
      const name = axios.get('/name/'+this.props.name)
        .then((res)=>{
          this.setState({
            fullname: res.data,
          })
        });
    }


    togglecv = () => {
        this.setState({
            cv: !this.state.cv,
            personalproj: false,
        })
    }

    togglepersonalproj = () => {
        this.setState({
            cv: false,
            personalproj: !this.state.personalproj
        })
    }


    render() {
        return (
                <div className="profile">
                <div className="profileRow">
                <Profile img = {'/image/'+this.props.name+'.jpg'} clientname = {this.state.fullname} colourMode = {this.props.colourMode}/>

                <div className="profileColumn">
                <div className="profileInfo">
                  <UserInfo clientname = {this.props.name} type = "description" colourMode = {this.props.colourMode}/>
                  <UserInfo clientname = {this.props.name} type = "skills" colourMode = {this.props.colourMode}/>
                  <UserInfo clientname = {this.props.name} type = "email" colourMode = {this.props.colourMode}/>
                </div>

                        <div className={"buttondiv"+this.props.colourMode}>
                          <button type="submit" testID="leftCVbutton" className={"btn"+this.props.colourMode+ " btn-primary"+this.props.colourMode+ " button"+this.props.colourMode} onClick = {this.togglecv}  style={{cursor:'pointer'}}>See CV</button>
                          <button type="submit" testID="leftProjectbutton" className={"btn"+this.props.colourMode+ " btn-primary"+this.props.colourMode+ " button"+this.props.colourMode} onClick = {this.togglepersonalproj} style={{cursor:'pointer'}}>See Personal Projects</button>
                        </div>
                  </div>
                  </div>

                  {this.state.cv &&
                    <div className = {"box"+this.props.colourMode}>
                    <img src={"/image/"+this.props.name+"Resume.jpg"}
                        className="cvPhoto" alt={this.props.name+"CV"}/>
                    </div>
                  }

                {this.state.personalproj &&
                    <div className = {"box"+this.props.colourMode}>
                    <ProjectsPage author = {this.props.author} colourMode = {this.props.colourMode}/>
                    </div>}
              </div>


        )
    }
}
