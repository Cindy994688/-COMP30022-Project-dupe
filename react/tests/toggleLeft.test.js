/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import ToggleLeft from '../src/pages/ToggleLeft';

test('it renders correctly', () => {
    component = renderer.create(<ToggleLeft name="ronchiu"/>)
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('it expands after the cv button is clicked', () => {
    component.root.findByProps({testID:"leftCVbutton"}).props.onClick();
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('it expands after the project button is clicked', () => {
    component.root.findByProps({testID:"leftProjectbutton"}).props.onClick();
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});


