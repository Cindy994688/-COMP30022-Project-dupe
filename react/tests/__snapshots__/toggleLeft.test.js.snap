// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it expands after the cv button is clicked 1`] = `
Array [
  <div
    className="topTriangleundefined container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-6"
      >
        <div
          className="profileImage"
        >
          <Image
            source="/image/ronchiu.jpg"
            style={
              Object {
                "borderRadius": 150,
                "height": 300,
                "width": 300,
              }
            }
          />
          <div
            className="profileNameundefined"
          >
            
          </div>
        </div>
      </div>
      <div
        className="col-md-6"
      >
        <div
          className="bioTextundefined"
        >
          
        </div>
        <div
          className="bioTextundefined"
        >
          
        </div>
        <div
          className="bioTextundefined"
        >
          
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="buttondivundefined"
        >
          <button
            className="btnundefined btn-primaryundefined buttonundefined"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
            testID="leftCVbutton"
            type="submit"
          >
            See CV
          </button>
          <button
            className="btnundefined btn-primaryundefined buttonundefined"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
            testID="leftProjectbutton"
            type="submit"
          >
            See Personal Projects
          </button>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="boxundefined"
  >
    <Image
      source="/image/MengyanResume.jpg"
      style={
        Object {
          "height": 900,
          "width": 636,
        }
      }
    />
  </div>,
]
`;

exports[`it expands after the project button is clicked 1`] = `
<div
  className="topTriangleundefined container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    >
      <div
        className="profileImage"
      >
        <Image
          source="/image/ronchiu.jpg"
          style={
            Object {
              "borderRadius": 150,
              "height": 300,
              "width": 300,
            }
          }
        />
        <div
          className="profileNameundefined"
        >
          
        </div>
      </div>
    </div>
    <div
      className="col-md-6"
    >
      <div
        className="bioTextundefined"
      >
        
      </div>
      <div
        className="bioTextundefined"
      >
        
      </div>
      <div
        className="bioTextundefined"
      >
        
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="buttondivundefined"
      >
        <button
          className="btnundefined btn-primaryundefined buttonundefined"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
          testID="leftCVbutton"
          type="submit"
        >
          See CV
        </button>
        <button
          className="btnundefined btn-primaryundefined buttonundefined"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
          testID="leftProjectbutton"
          type="submit"
        >
          See Personal Projects
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders correctly 1`] = `
<div
  className="topTriangleundefined container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    >
      <div
        className="profileImage"
      >
        <Image
          source="/image/ronchiu.jpg"
          style={
            Object {
              "borderRadius": 150,
              "height": 300,
              "width": 300,
            }
          }
        />
        <div
          className="profileNameundefined"
        >
          
        </div>
      </div>
    </div>
    <div
      className="col-md-6"
    >
      <div
        className="bioTextundefined"
      >
        
      </div>
      <div
        className="bioTextundefined"
      >
        
      </div>
      <div
        className="bioTextundefined"
      >
        
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="buttondivundefined"
      >
        <button
          className="btnundefined btn-primaryundefined buttonundefined"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
          testID="leftCVbutton"
          type="submit"
        >
          See CV
        </button>
        <button
          className="btnundefined btn-primaryundefined buttonundefined"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
          testID="leftProjectbutton"
          type="submit"
        >
          See Personal Projects
        </button>
      </div>
    </div>
  </div>
</div>
`;
